mobile/lib/ui/collections/album/list_item.dart

```diff
@@ -0,0 +1,179 @@
+import "package:flutter/material.dart";
+import "package:intl/intl.dart";
+import "package:logging/logging.dart";
+import "package:photos/generated/l10n.dart";
+import "package:photos/models/collection/collection.dart";
+import "package:photos/models/file/file.dart";
+import "package:photos/models/selected_albums.dart";
+import "package:photos/services/collections_service.dart";
+import "package:photos/theme/ente_theme.dart";
+import "package:photos/ui/components/buttons/icon_button_widget.dart";
+import "package:photos/ui/viewer/file/no_thumbnail_widget.dart";
+import "package:photos/ui/viewer/file/thumbnail_widget.dart";
+
+class AlbumListItemWidget extends StatelessWidget {
+  final Collection collection;
+  final void Function(Collection)? onTapCallback;
+  final void Function(Collection)? onLongPressCallback;
+  final SelectedAlbums? selectedAlbums;
+
+  const AlbumListItemWidget(
+    this.collection, {
+    super.key,
+    this.onTapCallback,
+    this.onLongPressCallback,
+    this.selectedAlbums,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final textTheme = getEnteTextTheme(context);
+    final colorScheme = getEnteColorScheme(context);
+    const sideOfThumbnail = 60.0;
+
+    return GestureDetector(
+      onTap: () {
+        if (onTapCallback != null) {
+          onTapCallback!(collection);
+        }
+      },
+      onLongPress: () {
+        if (onLongPressCallback != null) {
+          onLongPressCallback!(collection);
+        }
+      },
+      behavior: HitTestBehavior.opaque,
+      child: ListenableBuilder(
+        listenable: selectedAlbums!,
+        builder: (context, _) {
+          final isSelected =
+              selectedAlbums?.isAlbumSelected(collection) ?? false;
+          return AnimatedContainer(
+            curve: Curves.easeOut,
+            duration: const Duration(milliseconds: 200),
+            decoration: BoxDecoration(
+              border: Border.all(
+                color: isSelected
+                    ? colorScheme.strokeMuted
+                    : colorScheme.strokeFainter,
+              ),
+              borderRadius: const BorderRadius.all(
+                Radius.circular(4),
+              ),
+            ),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Flexible(
```

Comment:
This subtree (from this `Flexible`) doesn't depend on the listenable. Can improve performance by passing this subtree as `child` of `ListenableBuilder`. See https://api.flutter.dev/flutter/widgets/ListenableBuilder-class.html